src/components/GitPanel.tsx

```diff
@@ -794,19 +794,30 @@ export class GitPanel extends React.Component<IGitPanelProps, IGitPanelState> {
    *
    * @param path - repository path
    */
-  private async _hasIdentity(path: string): Promise<void> {
-    // If the repository path changes, check the user identity
-    if (path !== this._previousRepoPath) {
+  private async _hasIdentity(path: string): Promise<string> {
```

Comment:
let's return `null` if no transient identity is set.

```suggestion
  private async _hasIdentity(path: string): Promise<string | null> {
```